src/Fixer/Strict/DeclareStrictTypesPositionFixer.php

```diff
@@ -0,0 +1,194 @@
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Move the strict type declaration to the configured location. Requires PHP >= 7.0.',
+            [
+                new VersionSpecificCodeSample(
+                    '<?php declare(strict_types=1);',
+                    new VersionSpecification(70000)
+                ),
+            ],
+            null
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // must ran before SingleBlankLineBeforeNamespaceFixer, BlankLineAfterOpeningTagFixer and DeclareEqualNormalizeFixer.
+        return 2;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return PHP_VERSION_ID >= 70000 && $tokens[0]->isGivenKind(T_OPEN_TAG);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return false;
+    }
+
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('relocate_to', 'Whether ``declare(strict_types=1)`` should be placed on "next" or "same" line, after the opening ``<?php`` tag.'))
+                ->setAllowedValues([self::LINE_NEXT, self::LINE_SAME])
+                ->setDefault(self::LINE_NEXT)
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        // check if the declaration is already done
+        $searchIndex = $tokens->getNextMeaningfulToken(0);
+        $sequence = $this->getDeclareStrictTypeSequence();
+        $sequenceLocation = $tokens->findSequence(array_merge($sequence, [new Token(';')]), $searchIndex, null, false);
+
+        if (null === $sequenceLocation) {
+            $sequenceLocation = $tokens->findSequence($sequence, $searchIndex, null, false);
+        }
+
+        if (null !== $sequenceLocation) {
+            $this->fixLocation($tokens, $sequenceLocation);
+        }
+    }
+
+    /**
+     * @return Token[]
+     */
+    protected function getDeclareStrictTypeSequence()
+    {
+        static $sequence = null;
+
+        // do not look for open tag, closing semicolon or empty lines;
+        // - open tag is tested by isCandidate
+        // - semicolon or end tag must be there to be valid PHP
+        // - empty tokens and comments are dealt with later
+        if (null === $sequence) {
+            $sequence = [
+                new Token([T_DECLARE, 'declare']),
+                new Token('('),
+                new Token([T_STRING, 'strict_types']),
+                new Token('='),
+                new Token([T_LNUMBER, '1']),
+                new Token(')'),
+            ];
+        }
+
+        return $sequence;
+    }
+
+    private function insertSequence(Tokens $tokens, array $sequence = null)
```

Comment:
I too don't mind the duplication here that much (since it's quite minimal). That said, it's a great candidate for reuse (since the code is doing basically the same thing in both places) – and any bugfixes will have to also be duplicated to both locations at present.